packages/ton-client/src/store.ts

```diff
@@ -0,0 +1,28 @@
+import { vec3 } from 'gl-matrix';
+
+export const TAP_STATES = [0.75, 1];
+export const SWIPE_STATES = [0, 1];
+export const PINCH_STATES = [0, 1];
+
+class Store {
+  tapState = TAP_STATES[0];
+  swipeState = SWIPE_STATES[0];
+  pinchState = PINCH_STATES[0];
+
+  state: vec3 = vec3.fromValues(
```

Comment:
It's probably better to use a class for the state here. 

eg:  
```
class TouchState {
  constructor(
      public tapState: float, 
      public swipeState: float, 
      public pinchState: float
  ) {}
}
```
Or can you explain what a vec3.normalize(state) means? xD

Reply:
I think I know where you're coming from: 
```
gl.uniform3fv(renderUniLocs.touchManipulationState, store.state);
```
Here you are passing a vec3. 
But you can also pass single floats as uniforms: 
```
gl.uniform1f(renderUniLocs.touchManipulationState, store.tapState);
...
```
Which is nicer because you also get more reasonable variable names in the shader or do not have to deconstruct them: 
`touchManipulationState.x` would become `tapState`

Reply:
yeah true, will probably do it like that.